pulser-core/pulser/sequence/_seq_drawer.py

```diff
@@ -487,6 +408,278 @@ def phase_str(phi: float) -> str:
+        sampling_rate,
+        draw_input,
+        draw_modulation,
+        draw_phase_curve,
+        draw_target_regions=False,
+    )
+
+    # Gather additional data
+    for ch, sch in seq._schedule.items():
+        interp_pts: defaultdict[str, list[list[float]]] = defaultdict(list)
+        target: dict[Union[str, tuple[int, int]], Any] = {}
+
+        for slot in sch:
+            if slot.ti == -1:
+                target["initial"] = slot.targets
+                continue
+            if slot.type == "target":
+                target[(slot.ti, slot.tf - 1)] = slot.targets
+                continue
+            if slot.type == "delay":
+                continue
+            pulse = cast(Pulse, slot.type)
+            for wf_type in ["amplitude", "detuning"]:
+                wf = getattr(pulse, wf_type)
+                if isinstance(wf, InterpolatedWaveform):
+                    pts = wf.data_points
+                    pts[:, 0] += slot.ti
+                    interp_pts[wf_type] += pts.tolist()
+
+        data[ch].target = target
+        if interp_pts:
+            data[ch].interp_pts = dict(interp_pts)
+
+    if hasattr(seq, "_measurement"):
+        data["measurement"] = seq._measurement
+
+    total_duration = data["total_duration"]
+    time_scale = 1e3 if total_duration > 1e4 else 1
+    t = np.arange(total_duration) / time_scale
+    final_t = t[-1]
+    t_min = -final_t * 0.03
+    t_max = final_t * 1.05
+
+    for ch, axes in ch_axes.items():
+        ch_obj = seq.declared_channels[ch]
+        ch_data = data[ch]
+        basis = ch_obj.basis
+        ys = ch_data.get_input_curves()
+        ys_mod = [()] * 3
+        yseff = [()] * 3
+        draw_output = draw_modulation and (
+            ch_obj.mod_bandwidth or not draw_input
+        )
+        if draw_output:
+            ys_mod = ch_data.get_output_curves(ch_obj)
+
+        if sampling_rate:
+            curves = ys_mod if draw_output else ys
+            yseff = ch_data.interpolate_curves(curves, sampling_rate)
+        ref_ys = [
+            list(chain.from_iterable(all_ys))
+            for all_ys in zip(ys, ys_mod, yseff)
+        ]
+        max_amp = np.max(ref_ys[0])
+        max_amp = 1 if max_amp == 0 else max_amp
+        amp_top = max_amp * 1.2
+        amp_bottom = min(0.0, *ref_ys[0])
+        # Makes sure that [-1, 1] range is always represented
+        det_max = max(*ref_ys[1], 1)
+        det_min = min(*ref_ys[1], -1)
+        det_range = det_max - det_min
+        det_top = det_max + det_range * 0.15
+        det_bottom = det_min - det_range * 0.05
+        ax_lims = [
+            (amp_bottom, amp_top),
+            (det_bottom, det_top),
+            (min(0.0, *ref_ys[2]), max(1.1, *ref_ys[2])),
+        ]
+        ax_lims = [ax_lims[i] for i in ch_data.curves_on_indices()]
+
         if draw_phase_area:
             top = False  # Variable to track position of box, top or center.
             print_phase = not draw_phase_curve and any(
```

Comment:
All the for loops with an if selecting slots that are `TimeSlots` of type "Pulse" can be replaced by a for loop over the `_targetSlot` of `ChannelSamples.slots`. The phase can be obtained by looking at `ChannelSamples.phase[slot.ti: slot.tf]`